### DISCUSSION TITLE
Question about Uncaught Exception from the example mesh1D.py

### DISCUSSION THREAD

**Discussion by CalebDmArcher** [usnistgov/fipy#951] [Q&A]
I am trying the last example from [examples/diffusion/mesh1D.py](https://www.ctcms.nist.gov/fipy/examples/diffusion/generated/examples.diffusion.mesh1D.html) which is the "Fully implicit solutions are not without their pitfalls, particularly in steady state..." section.

The code of the steady-state part got the error "RuntimeError: Factor is exactly singular" which comes from the code `DiffusionTerm(coeff=D).solve(var=phi)`. Is it because the latest version of the library can catch and raise exceptions for the cases where there can be more than one solution for the PDE?

(The attached file is what I actually used which got the exception)
[fipy_exmaple4.zip](https://github.com/usnistgov/fipy/files/12650047/fipy_exmaple4.zip)

**Comment by guyer** [comment]
Different solvers respond in different ways to PDEs with multiple solutions. The example should handle this better (#952).
